### PR TITLE
Modify upgrade deletion logic to delete by labels

### LINKED ISSUE #69: bug: delete - after job deletion also delete respective pods
When armada deletes jobs as part of the upgrade workflow, depending on when the job is deleted, pods which were created by Kubernetes to fulfill the job can be left behind -- the usual case for this is rapidly running armada right after a large run which creates many releases.  If the second run triggers lots of upgrade workflow while many jobs are still in flight, job pods will get left behind (even though the jobs themselves are gone).  This results in essentially duplicate jobs.

The correct behavior should be to augment the cleanup to also find all pods with the label (in the target namespace), for example:

```
job-name=nova-db-init
```

from the following describe:

```
$ kubectl describe pods/nova-db-init-6nzvj  -n openstack
Name:		nova-db-init-6nzvj
Namespace:	openstack
Node:		clcp-demo-2/
Labels:		controller-uid=82afdb8f-52ac-11e7-9768-5c838fe14e68
		**job-name=nova-db-init**
...
```

and delete those as well.

### PR DESCRIPTION
**Pull request by ghost**
-Pods are now deleted directly by label instead of deleteing job
-Solves the issue of Armada leaving dangling pods

**What is the purpose of this pull request?**:

**What issue does this pull request address?**: Fixes #69 

**Notes for reviewers to consider**:

**Specific reviewers for pull request**:
@gardlt @drewwalters96

### CODE REVIEW

**Comment by gardlt** [comment]
@theyer excellent work, just one thing. and it will be good.

File: armada/handlers/tiller.py [hunk 1]
```diff
@@ -153,8 +153,25 @@ def _pre_update_actions(self, actions, namespace):
                 if "job" in action_type:
                     LOG.info("Deleting %s in namespace: %s", name, namespace)
                     self.k8s.delete_job_action(name, namespace)
-                    continue
-                LOG.error("Unable to execute name: %s type: %s ", name, type)
+                elif "pod" in action_type:
+                    # create label_selector string
+                    label_selector = 'release_name={}'.format(release_name)
```
Comment (gardlt) [inline armada/handlers/tiller.py]: for this pod deletion logic is okay, but i would move it to its own func. i know i would like to use it in #105

Comment (theyer) [inline armada/handlers/tiller.py]: sounds good i'll implement that now

**Review by drewwalters96 — APPROVED** [review]

File: armada/handlers/tiller.py [hunk 2]
```diff
@@ -169,6 +168,29 @@ def _pre_update_actions(self, actions, namespace):
         except Exception:
             LOG.debug("PRE: Could not create anything, please check yaml")
 
+    def delete_resource(self, release_name, resource_name, resource_type,
```
Comment (theyer) [inline armada/handlers/tiller.py]: @gardlt something like this? leaves an easy opening for you to insert daemonset stuff

File: armada/handlers/tiller.py [hunk 3]
```diff
@@ -150,11 +150,10 @@ def _pre_update_actions(self, actions, namespace):
...
                 name = action.get("name")
                 action_type = action.get("type")
-                if "job" in action_type:
-                    LOG.info("Deleting %s in namespace: %s", name, namespace)
-                    self.k8s.delete_job_action(name, namespace)
-                    continue
-                LOG.error("Unable to execute name: %s type: %s ", name, type)
+                labels = action.get('labels', None)
+
+                self.delete_resource(release_name, name, action_type,
```
Comment (gardlt) [inline armada/handlers/tiller.py]: - sperate the job and pod deletion logic.

File: armada/handlers/tiller.py (hunk 2 above)
Comment (gardlt) [inline armada/handlers/tiller.py]: this is good.

File: armada/handlers/tiller.py (hunk 3 above)
Comment (theyer) [inline armada/handlers/tiller.py]: @gardlt done

File: armada/handlers/tiller.py (hunk 2 above)
Comment (theyer) [inline armada/handlers/tiller.py]: @gardlt did you want me to implement some sort of wait for the resource to be deleted?

Comment (theyer) [inline armada/handlers/tiller.py]: @gardlt latest commit implements logic to wait for the replacement pod to deploy and enter ready state before starting main deployment. thoughts?

**Review by gardlt — APPROVED** [review]

File: armada/handlers/k8s.py [hunk 4]
```diff
@@ -92,3 +94,30 @@ def delete_namespace_pod(self, name, namespace="default", body=None):
...
+        base_pod_pattern = re.compile('^(.+)-[a-zA-Z0-9]+$')
+        if not base_pod_pattern.match(old_pod_name):
+            LOG.error(
+                'Could not identify new pod after purging %s', old_pod_name)
+            return
+        pod_base_name = base_pod_pattern.match(old_pod_name).group(1)
+
+        new_pod_name = ''
+        w = watch.Watch()
+        for event in w.stream(self.client.list_namespaced_pod, namespace):
```
Comment (theyer) [inline armada/handlers/k8s.py]: @gardlt only thing I wanted to ask you about was giving some timeout here. i'm just worried if for some reason a job/service/whatever to redeploy the pod isn't running, the stream will sit open forever. the stream can take a timeout but I wasn't sure what value to use. maybe pass the timeout from the chart? and set some other default?

**Pull request merged**